src/Lorentz/Contracts/RegistryDAO.hs

```diff
@@ -0,0 +1,159 @@
+      push True
+    else do
+      push False
+  else do
+    -- submitted proposal size is bigger than max proposal size
+    drop; drop; drop
+    push False
+
+-- |  slash_value ( c * frozen / d), c = 1, d = 1
+-- slashScaleValue = c; slashDivideValue = d
+registryDaoRejectedProposalReturnValue
+  :: forall s k v store. 
+      (IsoRegistryDaoProposalMetadata k v
+      , DAO.StorageC store (RegistryDaoContractExtra k v) (RegistryDaoProposalMetadata k v)
+      )
+  =>  (DAO.Proposal (RegistryDaoProposalMetadata k v)) : store : s
+  :-> ("slash_amount" :! Natural) : s
+registryDaoRejectedProposalReturnValue = do
+  toField #pProposerFrozenToken
+  swap;
+  stToField #sExtra
+
+  getField #ceSlashScaleValue; dip swap; mul
+  swap; toField #ceSlashDivisionValue; swap; ediv
+  ifSome car $
+    push (0 :: Natural)
+  toNamed #slash_amount
+
+-- | Update the propose changes
+decisionLambda
+  :: forall k v s store.
+  ( DAO.StorageC store (RegistryDaoContractExtra k v) (RegistryDaoProposalMetadata k v)
+  -- shared
+  , IsoRegistryDaoProposalMetadata k v
+  , NicePackedValue (RegistryDaoProposalMetadata k v)
+  )
+  =>  DAO.Proposal (RegistryDaoProposalMetadata k v) : store : s
+  :-> List Operation : store : s
+decisionLambda = do
+  toField #pMetadata
+  caseT
+    ( #cNormalProposalType /-> do
+        getField #npAgoraPostId
+        swap; toField #npDiff
+        iter $ do
+          stackType @(RegistryUpdate k v : Natural : store : s)
+          constructT @(RegistryEntry v)
+            ( fieldCtor $ getField #ruNewValue
+            , fieldCtor $ duupX @2
+            , fieldCtor $ now
+            )
+          stackType @(RegistryEntry v : RegistryUpdate k v : Natural : store : s)
+          dip $ dip swap
+          dip $ toField #ruKey
+          some; swap;
+          duupX @3; stToField #sExtra;
+          stackType @((RegistryDaoContractExtra k v) : k : Maybe (RegistryEntry v) : store : Natural : s)
+          dig @2; dig @2; stUpdate #ceRegistry
+          stSetField #sExtra
+          swap
+
+        drop @Natural
+        nil
+    , #cConfigProposalType /-> do
+        duupX @2; stToField #sExtra;
+        duupX @2; toField #cpFrozenScaleValue; ifSome (setField #ceFrozenScaleValue) nop
+        duupX @2; toField #cpFrozenExtraValue; ifSome (setField #ceFrozenExtraValue)  nop
+        duupX @2; toField #cpSlashScaleValue; ifSome (setField #ceSlashScaleValue) nop
+        duupX @2; toField #cpSlashDivisionValue; ifSome (setField #ceSlashDivisionValue) nop
+        dig @1; toField #cpMaxProposalSize; ifSome (setField #ceMaxProposalSize) nop
+        stSetField #sExtra
+        nil
+    )
+
+
+config ::
+  ( IsoRegistryDaoProposalMetadata k v
+  , NicePackedValue (RegistryDaoProposalMetadata k v)
+  )
+  => DAO.Config (RegistryDaoContractExtra k v) (RegistryDaoProposalMetadata k v)
+config = DAO.defaultConfig
+  { DAO.cDaoName = "Registry DAO"
+  , DAO.cDaoDescription = [md|Registry DAO description|]
```

Comment:
It doesn't look like a good description, can you please write a sentence or two explaining what this DAO does?

Reply:
I used the spec description for this now.